From 5c1c1d2c7f2ef7e5c6314a1d2b68de4d0a1b5d44 Mon Sep 17 00:00:00 2001
From: Rezoss <rezoss@local>
Date: Mon, 13 Jul 2026 16:33:30 +0200
Subject: [PATCH] Force enable Apple AirDrop ProtocolX support

S26U OneUI9 ShareLive carries the Apple-device / AirDrop ProtocolX UI
and runtime paths. Keep the local ProtocolX gate forced on for this port
and prevent Samsung policy sync from clearing the Apple-device sharing
option. Mosey still needs to be present as a trusted system priv-app with
its daemon service for the backend to work.

diff --git a/smali/yf/l5.smali b/smali/yf/l5.smali
index 1111111..2222222 100644
--- a/smali/yf/l5.smali
+++ b/smali/yf/l5.smali
@@ -274,91 +274,11 @@
 .end method
 
 .method public final e(Z)Lzw/u;
     .locals 3
 
-    sget-object v0, Lge/d;->G:Lge/d;
-
-    iget-boolean v0, v0, Lge/d;->q:Z
-
-    if-eqz v0, :cond_0
-
-    iget-object v0, p0, Lyf/l5;->g:Lqo/a;
-
-    iget-object v0, v0, Lqo/a;->r:Ljava/lang/Object;
-
-    check-cast v0, Landroid/content/SharedPreferences;
-
-    const-string v1, "protocol_x_supported"
-
-    const/4 v2, 0x1
-
-    invoke-interface {v0, v1, v2}, Landroid/content/SharedPreferences;->getBoolean(Ljava/lang/String;Z)Z
-
-    move-result v0
-
-    if-eqz v0, :cond_0
-
-    new-instance v0, Ltf/e1;
-
-    const/16 v1, 0x1b
-
-    invoke-direct {v0, v1, p0}, Ltf/e1;-><init>(ILjava/lang/Object;)V
-
-    new-instance v1, Ljx/d2;
-
-    const/4 v2, 0x5
-
-    invoke-direct {v1, v2, v0}, Ljx/d2;-><init>(ILjava/lang/Object;)V
-
-    iget-object p0, p0, Lyf/l5;->c:Lag/f;
-
-    check-cast p0, Ltf/h0;
-
-    invoke-virtual {p0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
-
-    new-instance v0, Ltf/g0;
-
-    const/16 v2, 0xc
-
-    invoke-direct {v0, p0, v2}, Ltf/g0;-><init>(Ltf/h0;I)V
-
-    new-instance p0, Ljx/d2;
-
-    const/4 v2, 0x5
-
-    invoke-direct {p0, v2, v0}, Ljx/d2;-><init>(ILjava/lang/Object;)V
-
-    sget-object v0, Ltf/q;->C:Ltf/q;
-
-    invoke-virtual {p0, v0}, Lzw/u;->f(Lcx/e;)Lox/c;
-
-    move-result-object p0
-
-    new-instance v0, Lb6/a;
-
-    const/16 v2, 0xa
-
-    invoke-direct {v0, p1, v2}, Lb6/a;-><init>(ZI)V
-
-    invoke-static {v1, p0, v0}, Lzw/u;->w(Lzw/u;Lzw/u;Lcx/b;)Lzw/u;
-
-    move-result-object p0
-
-    return-object p0
-
-    :cond_0
-    if-eqz p1, :cond_1
-
-    sget-object p0, Lyc/g;->H:Lyc/e;
-
-    const-string p1, "[ProtocolXRepository] supportsProtocolX: policyPao.protocolXSupported=false"
-
-    invoke-virtual {p0, p1}, Lyc/g;->e(Ljava/lang/String;)V
-
-    :cond_1
-    sget-object p0, Ljava/lang/Boolean;->FALSE:Ljava/lang/Boolean;
+    sget-object p0, Ljava/lang/Boolean;->TRUE:Ljava/lang/Boolean;
 
     invoke-static {p0}, Lzw/u;->j(Ljava/lang/Object;)Ljx/d2;
 
     move-result-object p0
 
diff --git a/smali/yf/x3.smali b/smali/yf/x3.smali
index 3333333..4444444 100644
--- a/smali/yf/x3.smali
+++ b/smali/yf/x3.smali
@@ -482,11 +482,13 @@
 
     const-string v3, "protocol_x_supported"
 
+    const/4 v2, 0x1
+
     invoke-static {v1, v3, v2}, Lc4/n0;->t(Landroid/content/SharedPreferences;Ljava/lang/String;Z)V
 
-    if-nez v2, :cond_d
+    goto :cond_d
 
     iget-object v0, v0, Lyf/d4;->e:Lmf/j;
 
     iget-object v1, v0, Lmf/j;->b:Landroid/content/SharedPreferences;
 
